expansion.bzl

```diff
@@ -0,0 +1,561 @@
+"""
+Module containing functions that aid in environment variable expansion.
+
+This could be replaced with the same impl via Skylib dependency, if/when PR is merged.
+See https://github.com/bazelbuild/bazel-skylib/pull/486.
+"""
+
+_CONSIDERED_KEY_FORMATS = ("${}", "${{{}}}", "$({})")
+
+def _valid_char_for_env_var_name(char):
```

Comment:
Could you document the args and expected types here? I think it helps, especially with larger chunks of code.

Reply:
Here, and everywhere else there's a `def`.

Reply:
Done. Filled out doc strings for all the functions.